pennylane/qnode_new/qnode.py

```diff
@@ -0,0 +1,935 @@
+# Copyright 2018 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Quantum nodes
+=============
+
+**Module name:** :mod:`pennylane.qnode`
+
+.. currentmodule:: pennylane
+
+The :class:`~qnode.QNode` class represents *quantum nodes*,
+encapsulating a *quantum function* or :ref:`variational circuit <varcirc>`
+and the computational device it is executed on.
+
+The computational device is an instance of the :class:`~_device.Device`
+class, and can represent either a simulator or hardware device. They can be
+instantiated using the :func:`~device` loader. PennyLane comes included with
+some basic devices; additional devices can be installed as plugins
+(see :ref:`plugins` for more details).
+
+
+Quantum functions
+-----------------
+
+.. _quantumfunc:
+
+We use the term *quantum function* to refer both to the abstract mathematical function
+represented by the variational quantum circuit, and the concrete Python function that defines it.
+The latter must be of the following form:
+
+.. code-block:: python
+
+    def my_quantum_function(x, y, *, w=None):
+        qml.RZ(x, wires=0)
+        qml.CNOT(wires=[0,1])
+        qml.RY(y, wires=0)
+        qml.RY(w, wires=1)
+        return qml.expval(qml.PauliZ(0))
+
+Quantum functions are a restricted subset of Python functions, adhering to the following
+constraints:
+
+* The body of the function must consist of only supported PennyLane
+  :mod:`operations <pennylane.ops>`, one per line.
+
+.. note::
+
+    The quantum operations cannot be used outside of a quantum circuit function, as all
+    :class:`Operations <pennylane.operation.Operation>` require a QNode in order to perform queuing on initialization.
+
+* The function must always return either a single or a tuple of
+  *measured observable values*, by applying a :mod:`measurement function <pennylane.measure>`
+  to an :mod:`observable <pennylane.ops>`.
+
+.. note::
+
+    Measured observables **must** come after all other operations at the end
+    of the circuit function as part of the return statement, and cannot appear in the middle.
+
+* The quantum function can take two kinds of parameters: *positional* and *auxiliary*.
```

Comment:
These names are just placeholders, I'll do a search and replace if you suggest something else. I just wanted to make sure the nomenclature is consistent throughout the module.